.gitlab/internal_image_deploy/internal_image_deploy.yml

```
@@ -2,192 +2,120 @@
+    - export RELEASE_TAG="${CI_COMMIT_REF_SLUG}${RELEASE_TAG_SUFFIX}"
+    - export TMPL_SRC_IMAGE="v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}${TMPL_SRC_IMAGE_SUFFIX}"
+    # BUILD_TAG is always RELEASE_TAG
+    - export BUILD_TAG="${RELEASE_TAG}"
+    # Fetch Gitlab token
     - GITLAB_TOKEN=$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $GITLAB_TOKEN write_api) || exit $?; export GITLAB_TOKEN
+    # Set TMPL_SRC_REPO based on branch and release tag based on branch
     - if [ "$BUCKET_BRANCH" = "beta" ] || [ "$BUCKET_BRANCH" = "stable" ]; then TMPL_SRC_REPO="${TMPL_SRC_REPO}-release"; fi
     - |
       if [ "$BUCKET_BRANCH" = "nightly" ]; then
         RELEASE_TAG="${RELEASE_TAG}-${CI_COMMIT_SHORT_SHA}"
         TMPL_SRC_REPO="${TMPL_SRC_REPO}-nightly"
       fi
     - if [ "$BUCKET_BRANCH" = "dev" ]; then RELEASE_TAG="dev-${RELEASE_TAG}-${CI_COMMIT_SHORT_SHA}"; fi
-    - "dda inv pipeline.trigger-child-pipeline --project-name DataDog/images --git-ref master --timeout 3600 --variable IMAGE_VERSION --variable IMAGE_NAME --variable RELEASE_TAG --variable BUILD_TAG --variable TMPL_SRC_IMAGE --variable TMPL_SRC_REPO --variable TMPL_ADP_VERSION --variable RELEASE_STAGING --variable RELEASE_PROD --variable DYNAMIC_BUILD_RENDER_RULES --variable APPS --variable BAZEL_TARGET --variable DDR --variable DDR_WORKFLOW_ID --variable TARGET_ENV --variable DYNAMIC_BUILD_RENDER_TARGET_FORWARD_PARAMETERS"
+    - "dda inv pipeline.trigger-child-pipeline --project-name DataDog/images --git-ref master --timeout 3600 --variable IMAGE_VERSION --variable IMAGE_NAME --variable RELEASE_TAG --variable TMPL_SRC_IMAGE --variable TMPL_SRC_REPO --variable TMPL_ADP_VERSION --variable RELEASE_STAGING --variable RELEASE_PROD --variable DYNAMIC_BUILD_RENDER_RULES --variable APPS --variable BAZEL_TARGET --variable DDR --variable DDR_WORKFLOW_ID --variable TARGET_ENV --variable DYNAMIC_BUILD_RENDER_TARGET_FORWARD_PARAMETERS"
   retry: 2
 
-docker_trigger_internal-fips:
-  stage: internal_image_deploy
-  rules: !reference [.on_deploy_internal_or_manual]
+# -- binary specific variables --
+.agent_variables: &agent_variables
+  IMAGE_VERSION: tmpl-v14
+  IMAGE_NAME: datadog-agent
+  TMPL_SRC_REPO: ci/datadog-agent/agent
+
+.cluster_agent_variables: &cluster_agent_variables
+  IMAGE_VERSION: tmpl-v6
+  IMAGE_NAME: datadog-cluster-agent
+  TMPL_SRC_REPO: ci/datadog-agent/cluster-agent
+  RELEASE_PROD: "true"
+
+.ot_standalone_variables: &ot_standalone_variables
+  IMAGE_VERSION: tmpl-v1
+  IMAGE_NAME: otel-agent
+  TMPL_SRC_REPO: ci/datadog-agent/otel-agent
+
+# -- publish jobs --
+publish-internal-container-image-jmx:
+  extends: .docker_trigger_base
+  needs:
+    - job: docker_build_agent7_jmx
+      artifacts: false
+    - job: docker_build_agent7_jmx_arm64
+      artifacts: false
+  variables:
+    <<: *agent_variables
+    TMPL_ADP_VERSION: 0.1.7
+    RELEASE_TAG_SUFFIX: "-jmx"
+    TMPL_SRC_IMAGE_SUFFIX: "-7-jmx"
+
+publish-internal-container-image-fips:
```

```suggestion
publish_internal_image-fips:
```
Here the last separator is well `-` as it specifies a `type`/`arch`/`os`